### PR TITLE
plugin/evm: Add ExampleWarp tests to vm warp unit tests

### PR DESCRIPTION
**Pull request by aaronbuchwald**
This PR adds new unit tests using the `ExampleWarp.sol` contract to validate that both payload types (AddressedPayload and BlockHashPayload) are handled correctly both when valid/invalid.

### CODE REVIEW

**Review by darioush — APPROVED** [review]

File: plugin/evm/vm_warp_test.go [hunk 1]
```diff
@@ -25,14 +26,23 @@ import (
...
 	"github.com/stretchr/testify/require"
 )
 
+var (
+	//go:embed ExampleWarp.bin
+	exampleWarpBin string
+	//go:embed ExampleWarp.abi
+	exampleWarpABI string
+)
+
```
Comment (anusha-ctrl) [inline plugin/evm/vm_warp_test.go:45]: nice

File: plugin/evm/vm_warp_test.go [hunk 2]
```diff
@@ -120,6 +130,262 @@ func TestSendWarpMessage(t *testing.T) {
...
+	signedMessage, err := avalancheWarp.NewMessage(
+		unsignedMessage,
+		warpSignature,
+	)
+	require.NoError(err)
+
+	createTx, err := types.SignTx(
+		types.NewContractCreation(0, common.Big0, 7_000_000, big.NewInt(225*params.GWei), common.Hex2Bytes(exampleWarpBin)),
+		types.LatestSignerForChainID(vm.chainConfig.ChainID),
```
Comment (anusha-ctrl) [inline plugin/evm/vm_warp_test.go:312]: can we have these as constants?

File: plugin/evm/vm_warp_test.go [hunk 3]
```diff
@@ -120,6 +130,262 @@ func TestSendWarpMessage(t *testing.T) {
...
+	require.Len(verifiedMessageReceipts, 2)
+	for i, receipt := range verifiedMessageReceipts {
+		require.Equal(types.ReceiptStatusSuccessful, receipt.Status, "index: %d", i)
+	}
+
+	tracerAPI := tracers.NewAPI(vm.eth.APIBackend)
+	txTraceResults, err := tracerAPI.TraceBlockByHash(context.Background(), ethBlock.Hash(), nil)
+	require.NoError(err)
+	require.Len(txTraceResults, 2)
+	blockTxTraceResultBytes, err := json.Marshal(txTraceResults[1].Result)
```
Comment (anusha-ctrl) [inline plugin/evm/vm_warp_test.go:376]: whats in txTraceResults[0]

File: contracts/contracts/ExampleWarp.sol [hunk 4]
```diff
@@ -64,7 +64,7 @@ contract ExampleWarp {
         uint256 index
     ) external view {
         (WarpBlockHash memory warpBlockHash, bool valid) = warp.getVerifiedWarpBlockHash(index);
-        require(valid);
+        require(!valid);
         require(warpBlockHash.sourceChainID == bytes32(0));
```
Comment (anusha-ctrl) [inline contracts/contracts/ExampleWarp.sol:68]: why did this change?

Comment (aaronbuchwald) [reply contracts/contracts/ExampleWarp.sol:68]: It was incorrect and unused before

File: plugin/evm/vm_warp_test.go (hunk 2 above)
Comment (aaronbuchwald) [reply plugin/evm/vm_warp_test.go:312]: Three is the magic number, so going to leave this as is for now.

File: plugin/evm/vm_warp_test.go [hunk 5]
```diff
@@ -120,6 +130,262 @@ func TestSendWarpMessage(t *testing.T) {
 	require.True(bls.Verify(vm.ctx.PublicKey, blsSignature, unsignedMessage.Bytes()))
 }
 
+func TestValidateWarpMessage(t *testing.T) {
+	require := require.New(t)
+	sourceChainID := ids.GenerateTestID()
+	sourceAddress := common.HexToAddress("0x376c47978271565f56DEB45495afa69E59c16Ab2")
+	destinationChainID := ids.GenerateTestID()
+	destinationAddress := common.HexToAddress("0x376c47978271565f56DEB45495afa69E59c16Ab2")
```
Comment (ceyonur) [inline plugin/evm/vm_warp_test.go:138]: does this need to be the same address as above?

**Review by ceyonur — APPROVED** [review]
just nit

**Pull request merged**